cpmpy/transformations/int2bool.py

```diff
@@ -0,0 +1,469 @@
+                rhs,
+                encoding,
+                weights=lhs.args[0],
+            )
+        else:
+            raise NotImplementedError(
+                f"int2bool: comparison with lhs {lhs} not (yet?) supported"
+            )
+
+    else:
+        raise NotImplementedError(
+            f"int2bool: non-comparison {expr} not (yet?) supported"
+        )
+
+
+def _encode_int_var(ivarmap, x, encoding):
+    """Return encoding of integer variable `x` and its domain constraints (if newly encoded)."""
+    if isinstance(x, (BoolVal, _BoolVarImpl)):
+        raise TypeError
+    elif x.name in ivarmap:  # already encoded
+        return ivarmap[x.name], []
+    else:
+        if encoding == "direct":
+            ivarmap[x.name] = IntVarEncDirect(x)
+        elif encoding == "order":
+            ivarmap[x.name] = IntVarEncOrder(x)
+        elif encoding == "binary":
+            ivarmap[x.name] = IntVarEncLog(x)
+        else:
+            raise NotImplementedError(encoding)
+
+        return (ivarmap[x.name], ivarmap[x.name].encode_domain_constraint())
+
+
+def _encode_linear(ivarmap, xs, cmp, rhs, encoding, weights=None, check_bounds=True):
+    """
+    Convert a linear constraint to a pseudo-boolean constraint.
+
+    Returns (newexpr, newcons)
+    """
+    if weights is None:
+        weights = len(xs) * [1]
+
+    # Check for trivial sat/unsat, since pysat does not handle those, and we avoid encoding them
+    if check_bounds is True:
+        # if the bounds are trivially sat/unsat, the value will be set to True/False
+        value = None
+        lb = sum(w * x.lb if w >= 0 else w * x.ub for w, x in zip(weights, xs))
+        ub = sum(w * x.ub if w >= 0 else w * x.lb for w, x in zip(weights, xs))
+
+        if cmp in ("==", "!="):
+            if lb == rhs == ub:
+                value = BoolVal(True)
+            elif rhs not in range(lb, ub + 1):
+                value = BoolVal(False)
+            if cmp == "!=" and value is not None:
+                value = ~value
+        elif cmp == "<=":
+            if ub <= rhs:
+                value = BoolVal(True)
+            elif lb > rhs:
+                value = BoolVal(False)
+        elif cmp == ">=":
+            if lb >= rhs:
+                value = BoolVal(True)
+            elif ub < rhs:
+                value = BoolVal(False)
+        else:
+            raise NotImplementedError
+
+        # if trivial sat/unsat, return early
+        if value is not None:
+            return [value], []
+
+    terms = []
+    domain_constraints = []
+    for w, x in zip(weights, xs):
+        # the linear may contain Boolean as well as integer variables
+        if isinstance(x, _BoolVarImpl):
+            terms += [(w, x)]
+        else:
+            x_enc, x_cons = _encode_int_var(
+                ivarmap, x, _decide_encoding(x, cmp, encoding)
+            )
+            domain_constraints += x_cons
+            new_terms, k = x_enc.encode_term(
+                w
```

Comment:
I find this multi-line for single-letter argument style surprising... you can just put the comment above the line of code, and then it is one nice compact line of code?

Reply:
True, that's my autoformatter, and putting the comment above fixes it.